src/wallet/rpcdump.cpp

```diff
@@ -138,6 +138,11 @@ UniValue importprivkey(const JSONRPCRequest& request)
         throw JSONRPCError(RPC_WALLET_ERROR, "Cannot import private keys to a wallet with private keys disabled");
     }
 
+    LegacyScriptPubKeyMan* spk_man = pwallet->GetLegacyScriptPubKeyMan();
+    if (!spk_man) {
+        throw JSONRPCError(RPC_WALLET_ERROR, "This type of wallet does not support this command");
```

Comment:
FWIW: I'm no longer in favor of splitting this out. For example this RPC method relies on `ImportPrivKeys` which is no longer a method on `CWallet`, but on`LegacyScriptPubKeyMan` (and not of its parent class `ScriptPubKeyMan`). This is in contrast to methods like `setwalletflag` that still call `CWallet` methods.